projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/JobSecretsService.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.secrets.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.vmware.taurus.base.FeatureFlags;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.vault.authentication.TokenAuthentication;
+import org.springframework.vault.client.VaultEndpoint;
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.support.Versioned;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class JobSecretsService {
+
+  private static final String SECRET = "secret";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+  private VaultTemplate vaultTemplate;
+
+  public JobSecretsService(
+      @Value("${vdk.vault.uri:}") String vaultUri,
+      @Value("${vdk.vault.token:}") String vaultToken,
+      FeatureFlags featureFlags)
+      throws URISyntaxException {
+
+    if (featureFlags.isVaultIntegrationEnabled()) {
```

Comment:
this seems crazy to use the feature flag when you could just use the property directly. 

also if it is disabled then the function updateJobSecrets just throws an error. 

I would add `@ConditionalOnProperty(value = "featureflag.vault.integration.enabled")` on top of this class and on top of the controller too.

Reply:
Fixed.